src/main.rs

```diff
@@ -53,7 +55,19 @@ fn main() {
         output: vec![fern::OutputConfig::stdout(), fern::OutputConfig::file("sccache2.log")],
         level: log::LogLevelFilter::Trace,
     };
-    if let Err(e) = fern::init_global_logger(logger_config, log::LogLevelFilter::Warn)
+
+    // Set current log level depending on environment variable
+    let log_level = match &*env::var("SCCACHE2_LOG_LEVEL").unwrap_or("warn".to_string()).to_lowercase() {
```

Comment:
Could you move this all into a separate function, like `fn init_logging()`, and just call that from main? I like having main with a very simple body.